src/Api/ProvidersApi.php

```diff
@@ -0,0 +1,90 @@
+<?php declare(strict_types = 1);
+
+namespace SandwaveIo\RealtimeRegister\Api;
+
+use SandwaveIo\RealtimeRegister\Domain\Downtime;
+use SandwaveIo\RealtimeRegister\Domain\DowntimeCollection;
+use SandwaveIo\RealtimeRegister\Domain\Provider;
+use SandwaveIo\RealtimeRegister\Domain\ProviderCollection;
+
+final class ProvidersApi extends AbstractApi
+{
+    /**
+     * @see https://dm.realtimeregister.com/docs/api/providers/get
+     *
+     * @param string $name
+     *
+     * @return Provider
+     */
+    public function get(string $name): Provider
+    {
+        $response = $this->client->get("/v2/providers/REGISTRY/{$name}");
+        return Provider::fromArray($response->json());
+    }
+
+    /**
+     * @see https://dm.realtimeregister.com/docs/api/providers/list
+     *
+     * @return ProviderCollection
+     */
+    public function list(
+        ?int $limit = null,
+        ?int $offset = null,
+        ?string $search = null
+    ): ProviderCollection {
+        $query = [];
+        if (! is_null($limit)) {
+            $query['limit'] = $limit;
+        }
+        if (! is_null($offset)) {
+            $query['offset'] = $offset;
+        }
+        if (! is_null($search)) {
+            $query['search'] = $search;
```

Comment:
As you we talked about, `$search` should be set on `$query['q']`

Reply:
Done: https://github.com/sandwave-io/realtimeregister-php/pull/22/commits/9bf827a5a24ca194f3bbba68b812273faecbe6ce